### PR TITLE
Fix: unet save_attn_procs at UNet2DconditionLoadersMixin

### PR DESCRIPTION
**Pull request by February24-Lee**
# What does this PR do?

An error occurs in the `save_function` when saving `safetensors` in the `save_attn_procs` of `UNet2DconditionLoadersMixin`. The error occurs when there are empty values in the `state_dict` called by `_get_custom_diffusion_state_dict()`.

### CODE REVIEW

File: src/diffusers/loaders/unet.py [hunk 1]
```diff
@@ -72,7 +72,11 @@ class UNet2DConditionLoadersMixin:
     unet_name = UNET_NAME
 
     @validate_hf_hub_args
-    def load_attn_procs(self, pretrained_model_name_or_path_or_dict: Union[str, Dict[str, torch.Tensor]], **kwargs):
+    def load_attn_procs(
```
Comment (sayakpaul) [inline src/diffusers/loaders/unet.py:75]: This shouldn't change.

File: src/diffusers/loaders/unet.py [hunk 2]
```diff
@@ -263,7 +267,10 @@ def _process_custom_diffusion(self, state_dict):
         for key, value_dict in custom_diffusion_grouped_dict.items():
             if len(value_dict) == 0:
                 attn_processors[key] = CustomDiffusionAttnProcessor(
-                    train_kv=False, train_q_out=False, hidden_size=None, cross_attention_dim=None
+                    train_kv=False,
+                    train_q_out=False,
+                    hidden_size=None,
+                    cross_attention_dim=None,
```
Comment (sayakpaul) [inline src/diffusers/loaders/unet.py:273]: Same. This formatting shouldn't change.

File: src/diffusers/loaders/unet.py [hunk 3]
```diff
@@ -451,12 +458,24 @@ def save_attn_procs(
...
         if is_custom_diffusion:
             state_dict = self._get_custom_diffusion_state_dict()
+            if save_function is None and safe_serialization:
+                # safetensors does not support saving dicts with non-tensor values
+                empty_state_dict = {k: v for k, v in state_dict.items() if not isinstance(v, torch.Tensor)}
+                logger.warning(
+                    f"Safetensors does not support saving dicts with non-tensor values. "
+                    f"The following keys will be ignored: {empty_state_dict.keys()}"
+                )
+                state_dict = {k: v for k, v in state_dict.items() if isinstance(v, torch.Tensor)}
```
Comment (sayakpaul) [inline src/diffusers/loaders/unet.py:478]: This is the only change that is relevant.

**Review by sayakpaul — COMMENTED** [review]
Thanks for your PR. Left some minor comments.

**Comment by February24-Lee** [comment]
@sayakpaul restored unchanged parts and modified the logging condition. ty.

**Comment by sayakpaul** [comment]
Thank you for your contributions. Good job!

**Pull request merged**